### PR TITLE
fix sub implementation in link_merging.asm test

### PR DESCRIPTION
**Pull request by onurinanc**
The previous version of link_merging.asm is not working correctly since the `operation sub<1>` implementation is wrong. With this PR, `operation sub<1>` implementation is fixed.

### CODE REVIEW

File: test_data/asm/permutations/link_merging.asm [hunk 1]
```diff
@@ -29,7 +29,7 @@ machine Main {
 
     // these are merged into 1 link
     instr add X, Y -> Z link => Z = submachine.add(X, Y);
-    instr sub_with_add X, Y -> Z link => Y = submachine.add(X, Z);
+    instr sub_with_add X, Y -> Z link => X = submachine.add(Y, Z);
```
Comment (leonardoalt) [inline test_data/asm/permutations/link_merging.asm:32]:
```suggestion
    instr sub_with_add X, Y -> Z link => Z = submachine.add(X, Y);
```
Shouldn't it be this?

Comment (onurinanc) [reply test_data/asm/permutations/link_merging.asm:32]: `instr sub_with_add X, Y -> Z` means that we want X - Y = Z (X = Y + Z). So, shouldn't we use `link => X = add(Y, Z)`

Comment (onurinanc) [reply test_data/asm/permutations/link_merging.asm:32]: It is not `instr add X, Y -> Z`, it is ìnstr sub_with_add`

Comment (leonardoalt) [reply test_data/asm/permutations/link_merging.asm:32]: ah right, I missed the sub part, agree with your PR

**Review by leonardoalt — DISMISSED** [review]

**Comment by leonardoalt** [comment]
looks like some tests need updating though

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**